src/common/colorspaces_inline_conversions.h

```diff
@@ -172,7 +172,7 @@ static inline void dt_apply_transposed_color_matrix(const dt_aligned_pixel_t in,
 {
   // using dt_aligned_pixel_t instead of float* for the function parameters gives GCC enough info to vectorize
   // and eliminate intermediate memory writes without going through major contortions
-  for_each_channel(r)
+  for(int r = 0; r < 3; r++)
```

Comment:
This will not only prevent the compiler from vectorizing the zeroing, it will also keep it from optimizing away memory accesses....  It would be better to find where the alpha channel is being set in such a way that it requires the color conversion code to preserve it, but I just spent an hour unsuccessfully trying to do so.  The obvious culprit of in-place color conversion in `dt_develop_blendif_lab_blend` isn't it, as even removing the conversion entirely didn't restore the yellow mask.

Reply:
I understand that this breaks the vectorizing, but the alpha channel is the 4th component here, you cannot set it to zero unconditionally. That's break the alpha channel. Or did I missed a change recently ? At least this patch works and it corresponds to what was done before your patch.

Reply:
I've found the real culprit.  The mask display only ever worked correctly as a side-effect of doing an in-place colorspace conversion, which the compiler would have complained about had `restrict` qualifiers been properly propagated up the call stack (`dt_ioppr_transform_image_colorspace` does not declare its input and output parameters to be `restrict`, which lets the code in pixelpipe_hb.c call it with the same buffer for both).  But  propagating the `restrict` will massively break the main pixelpipe code....

I have to clean things up for a PR (probably tomorrow), but the needed changes are to copy the alpha channel across the pixelwise conversions in `_transform_lab_to_rgb_matrix` (iop_profile.c) and `dt_develop_blendif_lab_blend` (blendif_lab.c).

Reply:
@ralfbrown : Ok, good finding... I'm then closing this PR.